### PR TITLE
CBL-815: test: add replicator goes offline scenario

### PR DESCRIPTION
**Pull request by jayahariv**
* most common scenario will be app goes background while data is transferring. make sure it will resume the transfer when it becomes online and continue

### CODE REVIEW

File: Objective-C/Tests/ReplicatorTest+Main.m [hunk 1]
```diff
@@ -463,6 +463,67 @@ - (void) testBackgroundingWhenStopping {
     r = nil;
 }
 
+- (void) testBackgroundingDuringDataTransfer {
+    CBLDatabase.log.console.level = kCBLLogLevelInfo;
```
Comment (jayahariv) [inline Objective-C/Tests/ReplicatorTest+Main.m:467]:
```suggestion
```

File: Objective-C/Tests/ReplicatorTest+Main.m [hunk 2]
```diff
@@ -463,6 +463,66 @@ - (void) testBackgroundingWhenStopping {
     r = nil;
 }
 
+- (void) testBackgroundingDuringDataTransfer {
+    XCTestExpectation* idle = [self allowOverfillExpectationWithDescription: @"idle-nd-ready"];
```
Comment (pasin) [inline Objective-C/Tests/ReplicatorTest+Main.m:467]: "idle-nd-ready" > "idle-and-ready"?

File: Objective-C/Tests/ReplicatorTest+Main.m [hunk 3]
```diff
@@ -463,6 +463,66 @@ - (void) testBackgroundingWhenStopping {
...
+    NSError* error;
+    CBLMutableDocument* doc1 = [[CBLMutableDocument alloc] initWithID: @"doc1"];
+    NSData* data = [self dataFromResource: @"image" ofType: @"jpg"];
+    CBLBlob* blob = [[CBLBlob alloc] initWithContentType: @"image/jpg" data: data];
+    [doc1 setBlob: blob forKey: @"blob"];
+    Assert([self.db saveDocument: doc1 error: &error]);
+    [self waitForExpectations: @[busy] timeout: 5.0];
+    
+    // background during the data transfer!
+    [replicator appBackgrounding];
```
Comment (pasin) [inline Objective-C/Tests/ReplicatorTest+Main.m:510]: What is the behavior of creating a background task when the device is not in background?

Comment (jayahariv) [reply Objective-C/Tests/ReplicatorTest+Main.m]: updated to use the setSuspended method

**Review by pasin — APPROVED** [review]

**Pull request merged**